### PR TITLE
ci: switch to GitHub actions from TravisCI

### PR DESCRIPTION
**Pull request by zchee**
Switch to GitHub actions from TravisCI. also, add codecov.

discuss: https://github.com/neovim/go-client/issues/39#issuecomment-626281917

---

Currently, the `nvim` packages test doesn't support Windows. I'll fix the testcase and adding `windows-latest` too.

And, IIRC @justinmk said "codecov doesn't need go-client because just neovim C API wrapper", but go-client have some Go related implements and testcase. So I introduced codecov to ensure Go side coverage.
@justinmk What do you think about it?

### CODE REVIEW

**Comment by h-michael** [comment]
FYI: There is github action for setupVim/Nvim.
https://github.com/rhysd/action-setup-vim

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    - name: Install nvim binary
+      uses: rhysd/action-setup-vim@v1
+      with:
+        neovim: true
+        version: nightly
+
+    - name: Test
+      run: |
+        go test -v -race -covermode=atomic -coverpkg=./... -coverprofile=coverage.out ./...
```
Comment (h-michael) [inline .github/workflows/test.yml:36]: @zchee 
When use Travis, we also ran "go vet", do we not need it?

Comment (zchee) [reply .github/workflows/test.yml:36]: ah, just forgot it.
will adding vet check.

**Comment by zchee** [comment]
@h-michael PTAL

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    - name: Install nvim binary
+      uses: rhysd/action-setup-vim@v1
+      with:
+        neovim: true
+        version: nightly
+
+    - name: Test and vet
+      run: |
+        diff -u <(echo -n) <(gofmt -s -d .)
+        go vet ./...
```
Comment (h-michael) [inline .github/workflows/test.yml:37]: Should `go vet ./... &&` ?

Comment (zchee) [reply .github/workflows/test.yml:37]: IIRC go vet will return exit 1 if failed. even if needs `&&` ?

Comment (h-michael) [reply .github/workflows/test.yml:37]: If `go vet is exit 1` and `go test` is exit 0, won't CI succeed?

Comment (zchee) [reply .github/workflows/test.yml:37]: IIRC, yes

**Pull request merged**